components/brave_news/browser/feed_v2_builder.cc

```diff
@@ -60,6 +66,36 @@ struct ArticleWeight {
 using ArticleInfo = std::tuple<mojom::FeedItemMetadataPtr, ArticleWeight>;
 using ArticleInfos = std::vector<ArticleInfo>;
 
+std::string GetFeedHash(const Channels& channels,
+                        const Publishers& publishers,
+                        const ETags& etags) {
+  std::vector<std::string> hash_items;
+  for (const auto& [channel_id, channel] : channels) {
+    if (!channel->subscribed_locales.empty()) {
+      hash_items.push_back(channel_id);
+    }
+  }
+
+  for (const auto& [id, publisher] : publishers) {
+    if (publisher->user_enabled_status == mojom::UserEnabled::ENABLED) {
+      hash_items.push_back(id);
+    }
+  }
+
+  for (const auto& [region, etag] : etags) {
+    hash_items.push_back(region + etag);
+  }
+
+  std::hash<std::string> hasher;
+  std::string hash;
+
+  for (const auto& hash_item : hash_items) {
+    hash = base::NumberToString(hasher(hash + hash_item));
```

Comment:
Q. It may sound dumb, I'm curious about this. is combining hash like this `new_hash_v = H(prev_hash_v + new_input)` used in general?

Reply:
I cargo culted this from the old feed system :laughing:

Reply:
Oh, I see. I was just curious if this should be something like `hasher(hash + hasher(hash_item))` so that we can lower the chance of collision, which I'm not sure about.